src/test/java/de/tum/in/www1/artemis/service/connectors/AtheneServiceTest.java

```diff
@@ -0,0 +1,265 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.test.util.ReflectionTestUtils;
+import org.springframework.test.web.client.ExpectedCount;
+import org.springframework.test.web.client.MockRestServiceServer;
+import org.springframework.web.client.RestTemplate;
+
+import de.tum.in.www1.artemis.AbstractSpringIntegrationBambooBitbucketJiraTest;
+import de.tum.in.www1.artemis.domain.*;
+import de.tum.in.www1.artemis.domain.enumeration.Language;
+import de.tum.in.www1.artemis.repository.*;
+import de.tum.in.www1.artemis.service.TextAssessmentQueueService;
+import de.tum.in.www1.artemis.service.TextBlockService;
+import de.tum.in.www1.artemis.service.TextSubmissionService;
+import de.tum.in.www1.artemis.util.ModelFactory;
+import de.tum.in.www1.artemis.web.rest.dto.AtheneDTO;
+
+public class AtheneServiceTest extends AbstractSpringIntegrationBambooBitbucketJiraTest {
+
+    @Autowired
+    RestTemplate restTemplate;
+
+    @Autowired
+    TextBlockService textBlockService;
+
+    @Autowired
+    TextAssessmentQueueService textAssessmentQueueService;
+
+    @Mock
+    TextExerciseRepository textExerciseRepository;
+
+    @Mock
+    TextBlockRepository textBlockRepository;
+
+    @Mock
+    TextSubmissionService textSubmissionService;
+
+    @Mock
+    TextClusterRepository textClusterRepository;
+
+    private final String API_ENDPOINT = "http://localhost/submit";
+
+    AtheneService atheneService;
+
+    TextExercise exercise1;
+
+    /**
+     * Initializes atheneService and example exercise
+     */
+    @BeforeEach
+    public void init() {
+        // Create atheneService and inject @Value fields
+        atheneService = new AtheneService(textSubmissionService, textBlockRepository, textBlockService, textClusterRepository, textExerciseRepository, textAssessmentQueueService);
+        ReflectionTestUtils.setField(atheneService, "ARTEMIS_SERVER_URL", ARTEMIS_SERVER_URL);
+        ReflectionTestUtils.setField(atheneService, "API_ENDPOINT", API_ENDPOINT);
+        String API_SECRET = "YWVuaXF1YWRpNWNlaXJpNmFlbTZkb283dXphaVF1b29oM3J1MWNoYWlyNHRoZWUzb2huZ2FpM211bGVlM0VpcAo=";
+        ReflectionTestUtils.setField(atheneService, "API_SECRET", API_SECRET);
+
+        // Create example exercise
+        ZonedDateTime pastTimestamp = ZonedDateTime.now().minusDays(5);
+        ZonedDateTime futureTimestamp = ZonedDateTime.now().plusDays(5);
+        Course course1 = ModelFactory.generateCourse(1L, pastTimestamp, futureTimestamp, new HashSet<>(), "tumuser", "tutor", "instructor");
+        course1.setRegistrationEnabled(true);
+        exercise1 = ModelFactory.generateTextExercise(pastTimestamp, futureTimestamp, futureTimestamp, course1);
+        exercise1.setId(1L);
+
+        when(textExerciseRepository.findWithEagerTeamAssignmentConfigAndCategoriesById(exercise1.getId())).thenReturn(Optional.ofNullable(exercise1));
```

Comment:
you don't need to mock a repository. just work with the actual database by saving the above exercise to the database!